StringOperator.java

```diff
@@ -0,0 +1,174 @@
+                if(!inword){
+                    inword = true;
+                    wordcount ++;
+                }
+            }else{
+                inword = false;
+            }
+        }
+        return wordcount;
+    }
+
+
+/***Checks if the current string is a palindrome. A palindrome is a word,
+phrase, or sequence that reads the same backward as forward.
+String text = "madam";
+boolean isPal = isPalindrome(); //
+ */  
+
+ 
+    public static String Palindrome(String input) {
+        char[] reverseString = new char[input.length()];
+        for (int i = 0; i < input.length(); i++) {
+            reverseString[i] = input.charAt(input.length() - 1 - i);
+        }
+        return new String(reverseString); // Return the reversed string
+    }
+
+/***Description: Reverses the current string.
+ * String reversedText = reverse(); // Result: "avaJ"
+ * 
+ */
+
+    public static String Reverse(String input) {
+        char[] reverseString = new char[input.length()];
+        for (int i = 0; i < input.length(); i++) {
+            reverseString[i] = input.charAt(input.length() - 1 - i);
+        }
+        return new String(reverseString);
+    }
+
+
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        Choose();
+        int choice;
+        try {
+            choice = scanner.nextInt();  // Read integer choice
+            scanner.nextLine(); // Consume newline after the integer input
+        } catch (Exception e) {
+            System.out.println("Invalid input. Please enter a number.");
+            scanner.close();
+            return; // Exit the program if input is invalid
+        }
+        
+        switch (choice) {
+            case 1:
+                String input;
+                do {
+                    System.out.println("Enter String To Append ('No' to exit): ");
+                    input = scanner.nextLine();
+                    if (!input.equalsIgnoreCase("No")) {
+                        currentString = Append_String(currentString, input);
+                        System.out.println("Updated String is: " + currentString);
+                    }
+                    System.out.println("If you want to append more then press Y else No: ");
+                    input = scanner.nextLine();
+                } while (input.equalsIgnoreCase("Y"));
+                break;
+
+
+
+            case 2:
+                System.out.println("Enter line of text (type 'NO' in new line to exit):");
+                while (true) {
+                    input = scanner.nextLine();
+                    if (input.equalsIgnoreCase("NO")) {
+                        break;
+                    }
+                    int wordcount = CountWords(input);
+                    System.out.println("Number of words: " + wordcount);
+                }
+                break;
+
+
+
+            case 4:
+                System.out.println("Enter any Word (Type 'No' to exit):");

```

Comment:
Properly closes the Scanner, which is good. Ensure that any additional resources (like file handles) are closed similarly in the future.